nixos/modules/system/activation/switch-to-configuration.py

```diff
@@ -0,0 +1,612 @@
+                    # FIXME: do something?
+                    pass
+                else:
+                    unit_info = parse_unit(new_unit_file)
+                    reload_if_changed = bool_is_true(unit_info.get('X-ReloadIfChanged', 'no'))
+                    restart_if_changed = bool_is_true(unit_info.get('X-RestartIfChanged', 'yes'))
+                    refuse_manual_stop = bool_is_true(unit_info.get('RefuseManualStop', 'no'))
+                    stop_if_changed = bool_is_true(unit_info.get('X-StopIfChanged', 'yes'))
+
+                    if reload_if_changed:
+                        transitions.reload_unit(unit)
+                    elif not restart_if_changed or refuse_manual_stop:
+                        transitions.skip_unit(unit)
+                    else:
+                        if not stop_if_changed:
+                            # This unit should be restarted instead of
+                            # stopped and started.
+                            transitions.restart_unit(unit)
+                        else:
+                            # If this unit is socket-activated, then stop the
+                            # socket unit(s) as well, and restart the
+                            # socket(s) instead of the service.
+                            socketActivated = False
+                            if unit.endswith('.service'):
+                                sockets = list(map(UnitName, unit_info.get('Sockets', '').split()))
+                                if len(sockets) == 0:
+                                    sockets = [UnitName("{base_name}.socket".format(base_name=base_name))]
+
+                                for socket in sockets:
+                                    if socket in active_prev:
+                                        transitions.stop_unit(socket)
+                                        transitions.start_unit(socket)
+                                        socket_activated = True
+
+                            # If the unit is not socket-activated, record
+                            # that this unit needs to be started below.
+                            # We write this to a file to ensure that the
+                            # service gets restarted if we're interrupted.
+                            if not socket_activated:
+                                transitions.start_unit(unit)
+
+                            transitions.stop_unit(unit)
+
+    return transitions
+
+def path_to_unit_name(path: Path) -> UnitName:
+    # Use current version of systemctl binary before daemon is reexeced.
+    args = ["/run/current-system/sw/bin/systemd-escape",
+            "--suffix=mount", "-p", str(path)]
+    output = subprocess.check_output(args)
+    return UnitName(output.strip())
+
+# Compare the previous and new fstab to figure out which filesystems
+# need a remount or need to be unmounted.  New filesystems are mounted
+# automatically by starting local-fs.target.  FIXME: might be nicer if
```

Comment:
This is already the case!

For every entry in `/etc/fstab` systemd will automatically generate  a `<device>.mount`  unit file.  Systemd doesn't use
`/etc/fstab` internally at all. It's all unit files

See `man systemd-fstab-generator`

Reply:
Also, if you type:

```
systemctl | grep mount
```
You'll see that there's a unit file per mount point present :)

Reply:
Oh, that fixme was copied from the previous code... ignore my dabbles :) We can fix that later.